workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs

```diff
@@ -42,17 +54,154 @@ public static void BuildConfig()
+                {
+                    var content = File.ReadAllText(latestLogFile.FullName);
+                    Debug.LogError($"{message}\n{content}");
+                }
+
+                process.Dispose();
+                process = null;
+            };
+        }
+
+        private static string GetClientLogFilename()
+        {
+            var logConfigPath = Path.Combine(SpatialProjectRootDir, "workers", "unity");
+            var configFileJson = File.ReadAllText(Path.Combine(logConfigPath, ClientConfigFilename));
+            var dict = Json.Deserialize(configFileJson);
+            var defaultLogFileName = "*unityclient.log";
+            Dictionary<string, object> tempDict;
+
+            var currentOS = "windows";
+            if (Application.platform == RuntimePlatform.OSXEditor)
+            {
+                currentOS = "macos";
+            }
+
+            if (!dict.TryGetValue("external", out var externalValue)) {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key 'external'.");
+                return defaultLogFileName;
+            }
+
+            tempDict = externalValue as Dictionary<string, object>;
+
+            if (!tempDict.TryGetValue("default", out var defaultValue))
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key 'default' within 'external'.");
+                return defaultLogFileName;
+            }
+
+            tempDict = defaultValue as Dictionary<string, object>;
+
+            if (!tempDict.TryGetValue(currentOS, out var currentOSValue))
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key '{currentOS}' within 'external' -> 'default'.");
+                return defaultLogFileName;
+            }
+
+            tempDict = currentOSValue as Dictionary<string, object>;
+
+            if (!tempDict.TryGetValue("arguments", out var argumentsValue))
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key 'arguments' within 'external' -> 'default' -> '{currentOS}'.");
+                return defaultLogFileName;
+            }
+
+            var arguments = (IList) argumentsValue;
+
+            int argumentIdx = 0;
+            for (argumentIdx = 0; argumentIdx < arguments.Count; argumentIdx++)
+            {
+                // The next argument would be the name of the log file
+                if (arguments[argumentIdx].Equals("-logfile"))
+                {
+                    argumentIdx++;
+                    break;
+                }
+            }
+
+            // Logger file not found - using default one
+            if (argumentIdx >= arguments.Count)
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain '-logfile' argument within 'external' -> 'default' -> '{currentOS}' -> arguments.");
+                return defaultLogFileName;
+            }
```

Comment:
Do you need the `.SkipWhile` if you have the `.FirstOrDefault`?

Reply:
Awesome, changed it.

Reply:
> 
> 
> Yes - otherwise you will return the first argument in the list that is not `-logfile`. So if the args were:
> 
> ```
>           "+workerType",
>           "UnityClient",
>           "+infraServicesUrl",
>           "http://127.0.0.1:21000",
>           "+projectName",
>           "${IMPROBABLE_PROJECT_NAME}",
>           "-logfile",
>           "../../logs/external-default-unityclient.log"
> ```
> 
> You will get `+workerType`

oooh... good one